### PR TITLE
Display correct IDs for openEO collections

### LINKED ISSUE #259: JSON endpoint lists empty ids
Our https://collections.eurodatacube.com/stac/index.json endpoint generates for some collections empty IDs. We should either remove them from the endpoint or make sure they are generated correctly (with full metadata).

Collections of questions:
- https://collections.eurodatacube.com/stac/phytoplankton-primary-production.json
- https://collections.eurodatacube.com/stac/co_3daily_data.json
- https://collections.eurodatacube.com/stac/s5p-ch4-weekly.json
- https://collections.eurodatacube.com/stac/so2_daily_data.json

### PR DESCRIPTION
**Pull request by horvatdino**
Closes #259 

This PR:
- Updates the following collections with the OpenEOPID:
  - `co_3daily_data`
  - `s5p-ch4_weekly`
  - `so2_daily_data`
- Removes `phytoplankton-primary-production` from the index endpoint

### CODE REVIEW

**Review by dthiex — CHANGES_REQUESTED** [review]

File: collections/co_3daily_data.yaml [hunk 1]
```diff
@@ -80,4 +81,4 @@ CubeDimensions:
     values:
       - co
```
[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/co_3daily_data.yaml:82]: You're right. Changed to `P1D`.

File: collections/so2_daily_data.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Name: SO2 daily data
+OpenEOPID: SENTINEL_5P_CH4_T1D
```
[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/so2_daily_data.yaml]: Removed.

File: collections/s5p-ch4-weekly.yaml [hunk 3]
```diff
@@ -80,4 +81,4 @@ CubeDimensions:
     values:
```
[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/s5p-ch4-weekly.yaml:81]: Waiting for the response.

[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/s5p-ch4-weekly.yaml:81]: Dorothy confirmed that this is a regularly updated collection, so this was updated to `null`.

**Review by dthiex — APPROVED** [review]

**Pull request merged**